lms/static/js/dashboard/views/course_view.js

```diff
@@ -0,0 +1,147 @@
+;(function (define) {
+    'use strict';
+
+    define(['jquery', 'backbone', 'underscore', 'logger'],
+        function ($, Backbone, _, Logger) {
+            var CourseView = Backbone.View.extend({
+
+                events: {
+                    'click .action-more': 'toggleCourseActionsDropdown',
+                    'click .action-unenroll': 'unEnroll',
+                    'click .action-email-settings': 'emailSettings',
+                    'click #upgrade-to-verified': 'upgradeToVerified',
+                    'click #block-course-msg a[rel="leanModal"]': 'unRegisterBlockCourse'
+                },
+
+                initialize: function (options) {
+                    this.setElement(this.el);
+                    this.listenTo(options.tabbedView, 'rendered', this.rendered);
+                },
+
+                rendered: function () {
+                    var $actionUnroll = this.$('.action-unenroll');
+                    var $unRegisterBlockCourse = this.$('#unregister_block_course');
+
+                    this.bindUnEnrollModal($actionUnroll);
+                    this.bindUnEnrollModal($unRegisterBlockCourse);
+                    this.bindEmailSettingsModal();
+                },
+
+                bindUnEnrollModal: function ($selector) {
+                    $selector.leanModal({
+                        overlay: 1,
+                        closeButton: ".close-modal"
+                    });
+
+                    var id = _.uniqueId('unenroll-');
+                    $selector.attr('id', id);
+                    var trigger = "#" + id;
+
+                    accessible_modal(
+                        trigger,
+                        "#unenroll-modal .close-modal",
+                        "#unenroll-modal",
+                        "#dashboard-main"
+                    );
+                },
+
+                bindEmailSettingsModal: function () {
+                    var $actionEmailSettings = this.$('.action-email-settings');
+
+                    $actionEmailSettings.leanModal({
+                        overlay: 1,
+                        closeButton: ".close-modal"
+                    });
+
+                    var id = _.uniqueId('email-settings-');
+                    $actionEmailSettings.attr('id', id);
+                    var trigger = "#" + id;
+
+                    accessible_modal(
+                        trigger,
+                        "#email-settings-modal .close-modal",
+                        "#email-settings-modal",
+                        "#dashboard-main"
+                    );
+                },
```

Comment:
The functionality of bindUnEnrollModal and bindEmailSettingsModal is very similar. You would be able to update to create a single bindModal function that accepts element and type as parameters to reduce code repetition. Assuming type for unenroll is the string ```unenroll``` you can use this to generate the id and append '-modal' and you have the id necessary for the accessible_modal function.